### DISCUSSION TITLE
Supporting different plugin formats

### DISCUSSION THREAD

**Discussion by adamski** [kunitoki/yup#3] [General]
I’m wondering about the best path to support various plugin formats, including AUv3 and AAX. 

- CLAP wrapper - currently limited to VST3 and AUv2
- Implement from scratch
- Borrow from / integrate with JUCE

**Comment by kunitoki** [comment]
Yes this is for sure planned. I would rather go there step by step, maybe we can initially have a look at the clap wrapper, but the long term goal is to implement them from scratch.

**Reply by kunitoki** [reply]
I started looking at this, i've tried clap-wrapper but it has some limitations, like for example is not possible to embed the clap plugin in a VST/AU, they have to scan for the clap library on disk which is far from being optimal. So i hacked around a work in progress [VST3 wrapper](https://github.com/kunitoki/yup/pull/44)

**Reply by kunitoki** [reply]
The basic structure for Clap, VST3 and standalone are up and running. Rough around the edges, i found out that it's the only way forward (clap wrapper was not very usable imho). Once i have some basic widgets running i will release the first official plugin made entirely with yup!

**Comment by kunitoki** [comment]
We now cover VST3, AU (v2 and v3), CLAP, AAX and LV2 with the last two having the plumbing already in place and requiring just finetuning.
